### PR TITLE
JAMES-3414 Email/set update: MailboxIds partial updates 

### PR DESCRIPTION
**Pull request by chibenwa**
I did not wrote tests for delegation as @LanKhuat  did test it on the base `reset mailboxIds pr` https://github.com/linagora/james-project/pull/3900

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Review by Arsnael — APPROVED** [review]

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailSetSerializer.scala [hunk 1]
```diff
@@ -1,34 +1,114 @@
-/****************************************************************
+/** **************************************************************
```
Comment (nvduc91) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailSetSerializer.scala:1]: Indents.

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/package.scala [hunk 2]
```diff
@@ -39,24 +39,30 @@ package object json {
...
+        mapWithStringKey
+          .foldLeft[Either[JsError, Map[K, V]]](firstAcc)((acc: Either[JsError, Map[K, V]], keyValue) => {
+            acc match {
+              case error@Left(_) => error
+              case scala.util.Right(validatedAcc) =>
+                val refinedKey: Either[String, K] = keyValidator.apply(keyValue._1)
+                refinedKey match {
+                  case Left(error) => Left(JsError(error))
+                  case scala.util.Right(unparsedK) =>
+                    val transformValue: JsResult[V] = valueReads.reads(keyValue._2)
```
Comment (nvduc91) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/package.scala:55]: `transformValue` already commented on my PR, should be `transformedValue`.

**Review by nvduc91 — APPROVED** [review]
You can merge and let me do fix this lite things

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**